concrete/controllers/panel/detail/page/composer.php

```diff
@@ -60,6 +60,21 @@ public function saveAndExit()
         }
     }
 
+    public function justSave()
+    {
+        if ($this->validateAction()) {
+            $r = $this->save();
+            $ptr = $r[0];
+            $u = new User();
+            $u->unloadCollectionEdit();
+            $c = $this->page;
+            $ptr->setRedirectURL(Core::make('helper/navigation')->getLinkToCollection($c, true));
+            $ptr->outputJSON();
+        } else {
+            throw new \Exception(t('Access Denied.'));
```

Comment:
Can you move this up as an early return since this logic actually happens first:

```php
if (!$this->validateAction() {
    throw new \Exception(...);
}

$result = $this->save();
$page = head($result);
```
etc

Reply:
If you'd like, I'm mostly worried about new code.